charts/consul/templates/mesh-gateway-deployment.yaml

```diff
@@ -179,6 +180,7 @@ spec:
       containers:
       - name: mesh-gateway
         image: {{ .Values.global.imageConsulDataplane | quote }}
+        {{- include "consul.restrictedSecurityContext" . | nindent 8 }}
```

Comment:
I'm not 100% sure this is safe for mesh-gateway. It can run in at least the cases I've tried, but I notice there is a PodSecurityPolicy that is conditional based on hostNetwork/hostPorts: https://github.com/hashicorp/consul-k8s/blob/8b45de8b153c158d68c972b9e872acc1e4b06825/charts/consul/templates/mesh-gateway-podsecuritypolicy.yaml#L1

I imagine this would fail if I enable hostNetwork or set hostPorts, but I don't understand those settings and need to test.

Reply:
I backed out the change to the mesh gateway deployment. Adding this "restricted" security context breaks mesh gateway whenever hostNetwork / hostPorts is enabled, with no workaround (i.e. cannot move it into a separate privileged namespace).

I think we'll need to create an SCC for mesh gateway when hostNetwork / hostPorts are enabled on OpenShift as well, but I don't have time to do this before the next release (PTO).